app/main.py

```diff
@@ -1,34 +1,95 @@
+from __future__ import annotations
+
+from typing import Tuple
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return f"({self.row}, {self.column})"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int],
+            end: Tuple[int],
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_decks()
+
+    def __repr__(self) -> str:
+        return f"Ship({self.start}, {self.end}, {self.is_drowned})"
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_decks(self) -> list[Deck]:
+        list_decks = []
+        for row in range(self.start[0], self.end[0] + 1):
+            for column in range(self.start[1], self.end[1] + 1):
+                deck = Deck(row=row, column=column)
+                list_decks.append(deck)
+        return list_decks
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck | None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
+
+    def fire(self, row: int, column: int) -> None:
+        fire_deck = self.get_deck(row=row, column=column)
+        if fire_deck:
+            fire_deck.is_alive = False
+            if all(not deck.is_alive for deck in self.decks):
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[Tuple[Tuple[int]]]) -> None:
```

Comment:
The type hint for `ships` should be `list[Tuple[Tuple[int, int], Tuple[int, int]]]` instead of `list[Tuple[Tuple[int]]]`. Each ship is defined by a start and end coordinate, both of which are tuples of two integers.